CMake/OtherTests.cmake

```diff
@@ -177,19 +177,20 @@ int main(void) {
   return 0;
 }" HAVE_STRUCT_TIMEVAL)
 
-
-include(CheckCSourceRuns)
-set(CMAKE_REQUIRED_FLAGS)
-if(HAVE_SYS_POLL_H)
-  set(CMAKE_REQUIRED_FLAGS "-DHAVE_SYS_POLL_H")
-endif(HAVE_SYS_POLL_H)
-check_c_source_runs("
+if(NOT ${CMAKE_CROSSCOMPILING})
+  include(CheckCSourceRuns)
+  set(CMAKE_REQUIRED_FLAGS)
+  if(HAVE_SYS_POLL_H)
+    set(CMAKE_REQUIRED_FLAGS "-DHAVE_SYS_POLL_H")
+  endif(HAVE_SYS_POLL_H)
+  check_c_source_runs("
   #ifdef HAVE_SYS_POLL_H
   #  include <sys/poll.h>
   #endif
   int main(void) {
     return poll((void *)0, 0, 10 /*ms*/);
   }" HAVE_POLL_FINE)
+endif()
```

Comment:
I'm honestly not 100% sure on this either; but it seemed like the attempt at running was a deliberate choice so I kept it. 

My assumption is that its testing for some particular form of poll and not just for the existence of poll itself. [It seems like non-zero return codes will fail the test](https://github.com/Kitware/CMake/blob/master/Modules/CheckCSourceRuns.cmake) even though they run, so I chalked it up to some platform somewhere not having a form of poll which works with the code.